new-docs/spelling/allowed_words.txt

```diff
@@ -0,0 +1,154 @@
+# Acronyms
```

Comment:
I think this should be a simple alphabetical list, and the `make` recipe should enforce this with `sort`. I don't see the categories as adding important information, and they make it harder to use than just sticking a word in the file.

Reply:
Removed the categories, added a Makefile target to sort the file, and sorted the file. Don't think it matters enough to enforce that the file is sorted in CI.